utils/is-store-name-valid.js

```diff
@@ -0,0 +1,13 @@
+function isStoreNameValid(storeName, storeNameConvention) {
```

Comment:
Maybe replace `storeNameConvention` by `context`? In this case, we can move validation here too, and simplify code of rules.